pkg/system/reconciler.go

```diff
@@ -307,6 +311,9 @@ func (r *Reconciler) CheckAll() {
 	util.KubeCheck(r.SecretServer)
 	util.KubeCheck(r.SecretOp)
 	util.KubeCheck(r.SecretEndpoints)
+	if len(r.SecretRootMasterKey) == 0 {
```

Comment:
but where do we set it? we are setting it only in phase2_creating ... am I missing something?

Reply:
There is a `SecretReconciler` interface:
https://github.com/baum/noobaa-operator/blob/0cceb836c8692b9127fafaa6701df35eac56c66b/pkg/util/kms/kms.go#L33-L40
Implemented by the system reconciler, the string value is set here:
https://github.com/baum/noobaa-operator/blob/0cceb836c8692b9127fafaa6701df35eac56c66b/pkg/system/phase2_creating.go#L956-L963
Invoked from here, by the "single string" KMS backends :
https://github.com/baum/noobaa-operator/blob/0cceb836c8692b9127fafaa6701df35eac56c66b/pkg/util/kms/kms_version.go#L34-L38

Reply:
again about this as r.SecretRootMasterKey is empty when creating a new reconciler - it will only be set when getting to phase2 (take a look we don't run the reconcile - we just create an instance of it in system.go:590). I don't think this check will help. Maybe we should use KubeCheckOptional and remove this if? WDYT? or check the KMS type somehow?

Reply:
Great point. I see no great value on reporting K8S secret existence, [exposed](https://github.com/noobaa/noobaa-operator/compare/83c199048e730e8ff771cf0b9f7d044829da4787..9c38cac0e8ffcca50c3a320deb09e2335eeeb3ca) KMS status condition instead. Thank you, Jacky 🖖